Composer/packages/client/src/pages/botProject/runtime-settings/RuntimeSettings.tsx

```diff
@@ -67,11 +77,13 @@ export const RuntimeSettings: React.FC<RouteComponentProps<{ projectId: string }
   const [templateKey, setTemplateKey] = useState('');
   const [runtimePath, setRuntimePath] = useState(settings.runtime?.path ?? '');
   const [runtimeCommand, setRuntimeCommand] = useState(settings.runtime?.command ?? '');
+  const [isAdaptive, setIsAdaptive] = useState(false);
   const [usingCustomRuntime, setUsingCustomRuntime] = useState(settings.runtime?.customRuntime ?? false);
 
   useEffect(() => {
     // check the status of the boilerplate material and see if it requires an update
     if (projectId) getBoilerplateVersion(projectId);
+    setIsAdaptive(isAdaptiveRuntime(settings));
```

Comment:
I think you want to useMemo here:
const isAdaptive = useMemo(isAdaptiveRuntime(settings),[settings]).

Otherwise you'll have to add settings to the dependency list of your effect.